app/src/main/java/com/duckduckgo/app/global/view/DaxDialog.kt

```diff
@@ -0,0 +1,147 @@
+    private val buttonText: String,
+    private val dismissible: Boolean = true,
+    private val typingDelayInMs: Long = 20
+) : DialogFragment() {
+
+    private var onAnimationFinished: () -> Unit = {}
+    private var primaryCTAClickListener: () -> Unit = { dismiss() }
+    private var hideClickListener: () -> Unit = { dismiss() }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
+        inflater.inflate(R.layout.content_dax_dialog, container, false)
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        val window = dialog.window
+        val attributes = window?.attributes
+
+        attributes?.gravity = Gravity.BOTTOM
+        window?.attributes = attributes
+        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+
+        return dialog
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setDialog()
+        setListeners()
+        dialogText.startTypingAnimation(daxText, true, onAnimationFinished)
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        dialogText.cancelAnimation()
+        super.onCancel(dialog)
+    }
+
+    fun onAnimationFinishedListener(onAnimationFinished: () -> Unit) {
+        this.onAnimationFinished = onAnimationFinished
+    }
+
+    fun setPrimaryCTAClickListener(clickListener: () -> Unit) {
+        primaryCTAClickListener = clickListener
+    }
+
+    fun setHideClickListener(clickListener: () -> Unit) {
+        hideClickListener = clickListener
+    }
+
+    fun startHighlightViewAnimation(targetView: View, duration: Long = 400, size: Float = 6f) {
+        val highlightImageView = addHighlightView(targetView, size)
+        val fadeInAnimation = ScaleAnimation(0f, 1f, 0f, 1f, Animation.RELATIVE_TO_SELF, 0.5f, Animation.RELATIVE_TO_SELF, 0.5f)
+        fadeInAnimation.duration = duration
+        fadeInAnimation.fillAfter = true
+        fadeInAnimation.interpolator = OvershootInterpolator(3f)
+        highlightImageView.startAnimation(fadeInAnimation)
+    }
+
+    private fun setListeners() {
+        hideText.setOnClickListener {
+            dialogText.cancelAnimation()
+            hideClickListener()
+        }
+
+        primaryCta.setOnClickListener {
+            dialogText.cancelAnimation()
+            primaryCTAClickListener()
+        }
+
+        if (dismissible) {
+            dialogContainer.setOnClickListener {
+                dialogText.cancelAnimation()
+                dismiss()
+            }
+        }
+    }
+
+    private fun setDialog() {
+        dialog?.window?.setLayout(ConstraintLayout.LayoutParams.MATCH_PARENT, ConstraintLayout.LayoutParams.MATCH_PARENT)
+        hiddenText.text = daxText
+        primaryCta.text = buttonText
+        dialogText.typingDelayInMs = typingDelayInMs
+    }
+
+    private fun addHighlightView(targetView: View, size: Float): View {
```

Comment:
I tried some large values for `size` here, expecting the circle to grow to silly sizes, but it seems to max out instead. Is that deliberate? If so, what's the restricting it... the parent view of the `targetView` maybe?

`dialog.startHighlightViewAnimation(targetView = gradeView, size = 500f)`

Reply:
It's actually the opposite, the smaller the number the bigger the circle it gets. This is related to the comment below about the kind of size you should provide. I'll try and make that a bit cleared.

Reply:
I have renamed `size` into `timesBigger` and refactor the code in order to hopefully have more clarity on what it does. This is a Float so you can specific how big the circle would be, the default value is 0 meaning that the circle will be 0 times bigger than the target view hence having the same size. A value of 1 will make the circle 1 times bigger (double the size) and so on so forth. In the specific case of the toolbar the values will have to be 0.Xf but I think at least that makes it more clear than `size`.

Reply:
I agree that it's an improvement over before. 

>0 times bigger than the target view hence having the same size
It still seems confusing, however, that to make something the same size you would choose 0.

I would expect to keep it the same size you would use 1 (since 1x anything gives the same value).
To make it twice the size, you would use 2 (since 2x anything doubles it) etc...
To make 3.5 times bigger, you would use 3.5

I don't want to dwell on that though, since it's not a showstopper and can be easily changed later.